src/pytorch_lightning/trainer/connectors/logger_connector/fx_validator.py

```diff
@@ -174,7 +174,7 @@ def check_logging(cls, fx_name: str) -> None:
             )
 
         if cls.functions[fx_name] is None:
-            raise MisconfigurationException(f"You can't `self.log()` inside `{fx_name}`.")
+            raise MisconfigurationException(f"You can alternatively log with `logger.experiment` inside `{fx_name}`.")
```

Comment:
I would keep the old message, making it now

```
You can't `self.log()` inside `a_hook`. HINT: You can still log directly to the logger by using `self.logger.experiment`
```

And then you can revert the changes you had to do to the tests